data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/client/MongoDBHelper.java

```diff
@@ -0,0 +1,156 @@
+import org.bson.types.Binary;
+import org.bson.types.Code;
+import org.bson.types.Decimal128;
+import org.bson.types.ObjectId;
+import org.bson.types.Symbol;
+import org.opensearch.dataprepper.plugins.mongo.configuration.MongoDBSourceConfig;
+import org.opensearch.dataprepper.plugins.truststore.TrustStoreProvider;
+
+import java.io.File;
+import java.util.Objects;
+
+import static com.mongodb.client.model.Filters.and;
+import static com.mongodb.client.model.Filters.gte;
+import static com.mongodb.client.model.Filters.lte;
+
+public class MongoDBHelper {
+    private static final String MONGO_CONNECTION_STRING_TEMPLATE = "mongodb://%s:%s@%s:%s/?replicaSet=rs0&directConnection=true&readpreference=%s&ssl=%s&tlsAllowInvalidHostnames=%s&directConnection=%s";
+    private static final String BINARY_PARTITION_FORMAT = "%s-%s";
+    private static final String BINARY_PARTITION_SPLITTER = "-";
+    private static final String TIMESTAMP_PARTITION_FORMAT = "%s-%s";
+    private static final String TIMESTAMP_PARTITION_SPLITTER = "-";
+
+    public static MongoClient getMongoClient(final MongoDBSourceConfig sourceConfig) {
+
+        final String connectionString = getConnectionString(sourceConfig);
+
+        final MongoClientSettings.Builder settingBuilder = MongoClientSettings.builder()
+            .applyConnectionString(new ConnectionString(connectionString));
+
+        if (Objects.nonNull(sourceConfig.getTrustStoreFilePath())) {
+            final File truststoreFilePath = new File(sourceConfig.getTrustStoreFilePath());
+            settingBuilder.applyToSslSettings(builder -> {
+                builder.enabled(sourceConfig.getSSLEnabled());
+                builder.invalidHostNameAllowed(sourceConfig.getSSLInvalidHostAllowed());
+                builder.context(TrustStoreProvider.createSSLContext(truststoreFilePath.toPath(),
+                        sourceConfig.getTrustStorePassword()));
+            });
+        }
+
+        return MongoClients.create(settingBuilder.build());
+    }
+
+    private static String getConnectionString(final MongoDBSourceConfig sourceConfig) {
+        final String username = sourceConfig.getCredentialsConfig().getUserName();
+        final String password = sourceConfig.getCredentialsConfig().getPassword();
+        final String hostname = sourceConfig.getHostname();
+        final int port = sourceConfig.getPort();
+        final String ssl = sourceConfig.getSSLEnabled().toString();
+        final String invalidHostAllowed = sourceConfig.getSSLInvalidHostAllowed().toString();
+        final String readPreference = sourceConfig.getReadPreference();
+        final String directionConnection = sourceConfig.getDirectConnection().toString();
+        return String.format(MONGO_CONNECTION_STRING_TEMPLATE, username, password, hostname, port,
+                readPreference, ssl, invalidHostAllowed, directionConnection);
+    }
+
+    public static String getPartitionStringFromMongoDBId(Object id, String className) {
```

Comment:
This public method is not used (except for tests). Do we need it?

Reply:
Yes, the partition creator will use this method. I will have this in following CR.